sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetFileFormat.scala

```diff
@@ -217,10 +217,14 @@ class GpuParquetFileFormat extends ColumnarFileFormat with Logging {
     val outputTimestampType = sparkSession.sessionState.conf.parquetOutputTimestampType
     val dateTimeRebaseException = "EXCEPTION".equals(
       sparkSession.sqlContext.getConf(ShimLoader.getSparkShims.parquetRebaseWriteKey))
+    // prior to spark 311 int96 don't check for rebase exception
+    // https://github.com/apache/spark/blob/068465d016447ef0dbf7974b1a3f992040f4d64d/sql/core/src/
+    // main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetWriteSupport.scala#L195
+    val isBeforeSpark311 = ShimLoader.getSparkShims.getSparkShimVersion.toString < "3.1.1"
```

Comment:
This has similar problems like I raised at https://github.com/NVIDIA/spark-rapids/pull/3330#discussion_r698554846.  A particular vendor flavor of Spark could have cherry-picked this behavior, so this needs to be a call _into_ the shim to check for the feature we're looking for rather than a direct check on the shim version string.